listenbrainz/webserver/views/playlist_api.py

```diff
@@ -918,21 +929,24 @@ def import_playlist_from_music_service(service):
     """
     user = validate_auth_header()
 
-    if service != "apple_music" and service != "spotify":
-        raise APIBadRequest(f"Service {service} is not supported. We currently only support 'spotify'.")
+    if service not in supported_services:
+        raise APIBadRequest(f"Service {service} is not supported. "
+                            f"Supported services are: 'spotify', 'apple_music', 'soundcloud'.")
+
+    service_class = supported_services[service]()
 
     if service == "spotify":
-        spotify_service = SpotifyService()
-        token = spotify_service.get_user(user["id"], refresh=True)
+        token = service_class.get_user(user["id"], refresh=True)
     elif service == "apple_music":
-        apple_service = AppleService()
         # TODO: implement refresh token for AppleMusic
-        token = apple_service.get_user(user["id"])
+        token = service_class.get_user(user["id"])
+    elif service == "soundcloud":
+        token = service_class.get_user(user["id"])
 
     if not token:
         raise APIBadRequest(f"Service {service} is not linked. Please link your {service} account first.")
 
-    if service == "apple_music" and not token["refresh_token"]:
+    if service == "apple_music" and not token.get("refresh_token"):
```

Comment:
I want to flag that this token check is not present in some of the other functions where the spotify token check that follows does appear.

Should the same check be added?
If so I will do some more refactoring to avoid duplicating the code again, and make a separate function to check the token.